### PR TITLE
renamed folders,files, setup input map, changes to movement

### PR DESCRIPTION
**Pull request by MrHasuu**
* changed folder names to match
* renamed some folders
* changed script names to lowercase snakecase
* set up project input mapping 
* minor change to the movement script

### CODE REVIEW

**Comment by MrHasuu** [comment]
@DissonantVoid i think godot doesnt like file/folder renaming/changing. lots of changes internally like the png.import files. and the scenes .tscn changes looks pretty bad. because git conflict would be nearly impossible since you dont actively change the code in a scene file. you wont be able to tell which one is the good change. 

also i noticed that my filename change for some stuff just isnt recognized that's a little frustrating.

renaming playermovement just deleted and added as new file too.

in conclusion. merge conflicts is gonna be a lot more painful than i thought down the line.

**Comment by DissonantVoid** [comment]
@MrHasuu I disagree, Godot is actually very compatible with version control and has a well organized file management, correct me if I'm wrong but I think what happened with you is that you moved some files in the file system and the .import files didn't move with them, this is to be expected as Godot does not track resources efficiently when moved outside the editor, the ideal way to move files is in the file viewer in the editor, moving a file there not only moves the .import file accordingly but also automatically updates all references so for example if you move a texture file every instance in every scene that uses that file will be updated automatically, so just avoid moving stuff outside the file view in the editor.

For merge conflicts I can't say that they won't occur because they sure will, but it's usually not that difficult to solve, the script files are obvious and the scenes (.tscn) files are also written in a human readable format and are really easy to understand, there is even some docs that can help you understand the format. I hope this clarifies things

**Comment by MrHasuu** [comment]
@DissonantVoid  i watched some videos on git + godot after. first thing that caught my attention was that he said you need to close and re-open godot every time you change branches because it retains some data from previous branch. this video is 3 years old, is this still the case? i think thats probably what happened with me.

his video also went into details on the scene files. it looks like it shouldnt be too bad, will take some time to get used to for me is all.

**Comment by DissonantVoid** [comment]
@MrHasuu I guess that's what happened with you, I don't think Godot would have a problem if the branch you changed to only has files that are new or modified, if a file is modified Godot will ask you if you want to "Reload" or "Resave". reloading is replacing the files that are open with the new files from the branch or the pulled commits which is usually what you want, resave is more of the "arrogant" choice, it's like saying I don't care about the new changes just rewrite them with the old changes again. I think problems occur when a file is deleted in the new update which can cause issues but this is just my theoretical knowledge and I haven't tested this. all in all I'd say listen to the video and close the editor when changing branches etc.. it's safer and causes less confusion to you and to the engine.
